bpf/lib/nodeport.h

```diff
@@ -1208,8 +1189,31 @@ static __always_inline bool snat_v4_needed(struct __ctx_buff *ctx, __be32 *addr,
 		if (info->sec_label == REMOTE_NODE_ID)
 			return false;
 #endif
+ #if defined(ENABLE_EGRESS_GATEWAY)
+		/* check egress gateway  policy only for traffic which is:
+		 *  - remote, i.e., not from a local endpoint (inc. local host)
+		 *  - is not destined for a remote node
+		 */
+		if (!ep || info->sec_label != REMOTE_NODE_ID) {
+			struct egress_info *einfo;
```

Comment:
The code and comment are consistent with the commit description, but I think the comment could use a bit more detail. Maybe something like:
```
/* Check egress gateway policy only for traffic which matches one of the
 * following conditions.
 *  - Not from a local endpoint (inc. local host): that tells us the
 *    traffic was redirected by an egress gateway policy to this node to
 *    be masqueraded.
 *  - Not destined for a remote node: that tells us the traffic is leaving
 *    the cluster. Inter-node traffic to remote pods would either leave
 *    through the tunnel or match the above IPV4_SNAT_EXCLUSION_DST_CIDR
 *    check.
 */
```

(Worth noting this assumes `IPV4_SNAT_EXCLUSION_DST_CIDR` will always be defined when both native routing and BPF masquerading are enabled, which I think is true.)

Reply:
added, thanks!